src/app/DIAmeterApplication.cpp

```diff
@@ -73,241 +80,871 @@ int DIAmeterApplication::main(const vector<string>& input_files, const string in
-		  int nCandPeptide = active_peptide_queue->SetActiveRange(min_mass, max_mass, min_range, max_range, candidatePeptideStatus);
-		  int candidatePeptideStatusSize = candidatePeptideStatus->size();
-		  carp(CARP_DETAILED_DEBUG, "nCandPeptide:%d \t candidatePeptideStatusSize:%d \t mass range:[%f,%f]", nCandPeptide, candidatePeptideStatusSize, min_range, max_range);
-		  if (nCandPeptide == 0) { continue; }
-
-		  TideMatchSet::Arr2 match_arr2(candidatePeptideStatusSize); // Scored peptides will go here.
-		  // Programs for taking the dot-product with the observed spectrum are laid
-		  // out in memory managed by the active_peptide_queue, one program for each
-		  // candidate peptide. The programs will store the results directly into
-		  // match_arr. We now pass control to those programs.
-		  TideSearchApplication::collectScoresCompiled(active_peptide_queue, spectrum, observed, &match_arr2, candidatePeptideStatusSize, charge);
-
-		  // The denominator used in the Tailor score calibration method
-		  double quantile_score = getTailorQuantile(&match_arr2);
-		  // carp(CARP_DETAILED_DEBUG, "Tailor quantile_score:%f", quantile_score);
-
-		  TideMatchSet::Arr match_arr(nCandPeptide);
-		  for (TideMatchSet::Arr2::iterator it = match_arr2.begin(); it != match_arr2.end(); ++it) {
-			  /// Yang: I cannot understand here that the peptide index and the rank is complement.
-			  /// More attention is needed.
-			  int peptide_idx = candidatePeptideStatusSize - (it->second);
-			  if ((*candidatePeptideStatus)[peptide_idx]) {
-				  TideMatchSet::Scores curScore;
-			      curScore.xcorr_score = (double)(it->first / XCORR_SCALING);
-			      curScore.rank = it->second;
-			      curScore.tailor = ((double)(it->first / XCORR_SCALING) + 5.0) / quantile_score;
-			      match_arr.push_back(curScore);
-			      // carp(CARP_DETAILED_DEBUG, "peptide_idx:%d \t xcorr_score:%f \t rank:%d", peptide_idx, curScore.xcorr_score, curScore.rank);
-			  }
-		  }
-		  TideMatchSet matches(&match_arr, highest_ms2_mz);
-		  if (!match_arr.empty()) { reportDIA(output_file, origin_file, *sc, active_peptide_queue, proteins, locations, &matches, &ms1scan_intensity_rank_map, &peptide_predrt_map); }
+  if (Params::GetBool("psm-filter")) {
+	  stringstream param_ss;
+	  param_ss << "diameter-search.filtered." << getCoeffTag() << ".txt";
+	  output_file_name_filtered_ = make_file_path(param_ss.str().c_str());
+  }
 
-	  }
 
-	  delete spectra;
-	  delete sc_index;
-	  for (map<int, pair<double*, double*>>::const_iterator i = ms1scan_intensity_rank_map.begin(); i != ms1scan_intensity_rank_map.end(); i++) {
-		  delete[] (i->second).first;
-		  delete[] (i->second).second;
+  // Extract all edge features
+  if (!FileUtils::Exists(output_file_name_unsorted_) || Params::GetBool("overwrite") /* */) {
+	  carp(CARP_DEBUG, "Either file exists or it needs to be overwrite:%s", output_file_name_unsorted_.c_str());
```

Comment:
grammar mistake - "overwrite" should be "overwritten". Also probably best to add a space after ":"